# DEAP Dataset Preliminary Exploration

In this notebook we intend to explore the preprocessed python files of the DEAP dataset and run some simple analyses

In [1]:
# DEAP preprocessed data construction
# Lets get a brief overview of one piece of data
# The data is cut up into 32 pieces each with their own dat file loadable via pickle

import cPickle
x = cPickle.load(open('deap_data/data_preprocessed_python/s01.dat', 'rb'))
print type(x)
print x['labels'].shape
print x['data'].shape

<type 'dict'>
(40, 4)
(40, 40, 8064)


The data looks as follows:

| Array name | Array shape |	Array contents |
| ----------- | ----------- | ------------------| 
| data | 40 x 40 x 8064 |	video/trial x channel x data | 
| labels | 40 x 4 |video/trial x label (valence, arousal, dominance, liking) |


1. The data was downsampled to 128Hz.
2. EOG artefacts were removed as in [1].
3. A bandpass frequency filter from 4.0-45.0Hz was applied.
4. The data was averaged to the common reference.
5. The EEG channels were reordered so that they all follow the Geneva order as above.
6. The data was segmented into 60 second trials and a 3 second pre-trial baseline removed.
7. The trials were reordered from presentation order to video (Experiment_id) order.